### PR TITLE
German prompts

### PR DESCRIPTION
**Pull request by ischender**
Moving most of the functions involving openAI to the German code, and translating prompts.

### CODE REVIEW

File: src/langcheck/metrics/de/reference_free_text_quality.py [hunk 1]
```diff
@@ -97,44 +95,135 @@ def sentiment(
     # The English prompt works well enough for German
     # TODO: Investigate the performance improvement with German prompt
```
Comment (yosukehigashi) [inline src/langcheck/metrics/de/reference_free_text_quality.py:96]: This is done!

File: src/langcheck/metrics/de/reference_free_text_quality.py [hunk 2]
```diff
@@ -97,44 +95,135 @@ def sentiment(
...
+    explanation_list = []
+    for gen in tqdm_wrapper(generated_outputs):
+        score, explanation = oai_evaluator.get_score(_prompt(gen_output=gen),
+                                                     _function_call_prompt)
+        score_list.append(score)
+        explanation_list.append(explanation)
+    return score_list, explanation_list
+
+
 def fluency(
```
Comment (yosukehigashi) [inline src/langcheck/metrics/de/reference_free_text_quality.py:227]: Fluency also has an OpenAI version, so for that one, we can pass in the German text directly.

File: src/langcheck/metrics/de/source_based_text_quality.py [hunk 3]
```diff
@@ -83,34 +83,139 @@ def factual_consistency(
     # The English prompt works well enough for German, like with Japanese
     # TODO: Investigate performance improvement with German prompt / translation
     if model_type == 'openai' or model_type == 'azure_openai':
```
Comment (yosukehigashi) [inline src/langcheck/metrics/de/source_based_text_quality.py:85]: This is done!

File: src/langcheck/metrics/de/reference_free_text_quality.py
Comment (yosukehigashi) [inline src/langcheck/metrics/de/reference_free_text_quality.py:1]: We recently added a new metric called `answer_relevance` in the English and Japanese versions, which just has an OpenAI version. So if you could also translate the prompt to german, that would be great!

**Review by yosukehigashi — COMMENTED** [review]
Thanks @ischender!

File: src/langcheck/metrics/de/reference_free_text_quality.py (hunk 1 above)
Comment (ischender) [reply src/langcheck/metrics/de/reference_free_text_quality.py:96]: True, to be honest, I did not investigate the improvements...

File: src/langcheck/metrics/de/reference_free_text_quality.py (hunk 2 above)
Comment (ischender) [reply src/langcheck/metrics/de/reference_free_text_quality.py:227]: ah yes, I missed that, adding that too...

**Review by yosukehigashi — COMMENTED** [review]
Thanks! Just a couple of docstring comments, but otherwise LGTM. I'll take a final look after #81 is merged first

File: src/langcheck/metrics/de/reference_free_text_quality.py [hunk 4]
```diff
@@ -94,78 +93,273 @@ def sentiment(
...
     openai_client: Optional[OpenAI] = None,
     openai_args: Optional[Dict[str,
                                str]] = None) -> MetricValue[Optional[float]]:
-    ''' We first translate the generated outputs to English, and then use the
-    Parrot fluency model to calculate the fluency scores, from the English
-    counterpart.
+    ''' If using the local models, we first translate the generated outputs
+    to English, and then use the Parrot fluency model to calculate the fluency
+    scores, from the English counterpart. If using the OpenAI models, we use
+    the OpenAI API to calculate the fluency scores. This metric takes on float
```
[the remark this answers is not in the document]
Comment (ischender) [reply src/langcheck/metrics/de/reference_free_text_quality.py:236]: good point, done it now

[the remark this answers is not in the document]
Comment (yosukehigashi) [reply src/langcheck/metrics/de/reference_free_text_quality.py]: Thanks!

**Review by yosukehigashi — APPROVED** [review]
LGTM! Excited to try this out on some German text (even though I won't understand it 😆). I'll merge this once tests pass!

**Pull request merged**